src/libs/blueprint/conduit_blueprint_mesh_partition.cpp

```diff
@@ -3352,55 +3352,86 @@ Partitioner::unstructured_topo_from_unstructured(const conduit::Node &n_topo,
     }
     else if(shape.is_polyhedral())
     {
-        conduit::Node n_indices;
-        as_index_t(n_topo["elements/connectivity"], n_indices);
-        auto iptr = as_index_t_array(n_indices);
+        const auto conn = n_topo["elements/connectivity"].as_index_t_accessor();
 
         // NOTE: I'm assuming that offsets are built already.
-        conduit::Node n_offsets, n_sizes;
-        as_index_t(n_topo["elements/offsets"], n_offsets);
-        as_index_t(n_topo["elements/sizes"], n_sizes);
-        auto offsets = as_index_t_array(n_offsets);
-        auto sizes = as_index_t_array(n_sizes);
-
-        conduit::Node n_se_conn, n_se_offsets, n_se_sizes;
-        as_index_t(n_topo["subelements/connectivity"], n_se_conn);
-        as_index_t(n_topo["subelements/offsets"], n_se_offsets);
-        as_index_t(n_topo["subelements/sizes"], n_se_sizes);
-        auto se_conn = as_index_t_array(n_se_conn);
-        auto se_offsets = as_index_t_array(n_se_offsets);
-        auto se_sizes = as_index_t_array(n_se_sizes);
+        const auto offsets = n_topo["elements/offsets"].as_index_t_accessor();
+        const auto sizes = n_topo["elements/sizes"].as_index_t_accessor();
+
+        const auto se_conn = n_topo["subelements/connectivity"].as_index_t_accessor();
+        const auto se_offsets = n_topo["subelements/offsets"].as_index_t_accessor();
+        const auto se_sizes = n_topo["subelements/sizes"].as_index_t_accessor();
+
+        // Figure out the index of each face being used. The first use of a face
+        // will put 0 in faceOrientation, the second use will be 1.
+        const auto maxFace = conn.max() + 1;
+        const auto connlen = conn.number_of_elements();
+        std::vector<int> faceCount(maxFace, 0);
+        std::vector<int> faceOrientation(connlen, 0);
+        for(index_t i = 0; i < connlen; i++)
+        {
+            const auto faceId = conn[i];
+            faceOrientation[i] = faceCount[faceId]++;
+        }
 
         std::map<index_t,index_t> old2new_faces;
         std::vector<index_t> new_sizes, new_offsets, new_se_conn, new_se_sizes, new_se_offsets;
 
         index_t new_offset = 0, new_se_offset = 0;
         for(auto eid : element_ids)
         {
-            auto offset = static_cast<index_t>(offsets[eid]);
-            auto nfaces = static_cast<index_t>(sizes[eid]);
+            const auto offset = static_cast<index_t>(offsets[eid]);
+            const auto nfaces = static_cast<index_t>(sizes[eid]);
```

Comment:
i think you can drop these static casts

Reply:
That's fair. I was just adding const to everything and didn't pay enough attention.